### PR TITLE
Exposing Request object in logger

### PR DESCRIPTION
**Pull request by busslina**

### CODE REVIEW

File: pkgs/shelf/lib/src/middleware/logger.dart [hunk 1]
```diff
@@ -19,7 +20,8 @@ import '../middleware.dart';
 /// The `isError` parameter indicates whether the message is caused by an error.
 ///
 /// If [logger] is not passed, the message is just passed to [print].
-Middleware logRequests({void Function(String message, bool isError)? logger}) =>
+Middleware logRequests(
```
Comment (kevmoo) [inline pkgs/shelf/lib/src/middleware/logger.dart:23]: I see where you're going here, but this is a BIG breaking change.

Comment (busslina) [reply pkgs/shelf/lib/src/middleware/logger.dart:23]: Why is it a big breaking change? Migrating from old to new code by adding one more argument.

Comment (natebosch) [reply pkgs/shelf/lib/src/middleware/logger.dart:23]: What is the motivation for this change? What else are you looking for that isn't in the `message`? Do you have an example logger callback you are planning to use with this signature?

Comment (busslina) [reply pkgs/shelf/lib/src/middleware/logger.dart:23]: My goal is to access some data stored on the Request context map and print info with the logger.

**Comment by natebosch** [comment]
Closing since I this is not a signature change we plan on making.

**Pull request closed without merging**